jsx/LanguageMenu.tsx

```diff
@@ -0,0 +1,76 @@
+import {useEffect, useState} from 'react';
+import {useTranslation} from 'react-i18next';
+import 'i18n';
+
+const supportedLanguageCodes = ['en', 'fr'];
+
+function filterSupportedLanguageCode(languageCode: string) {
+  if (supportedLanguageCodes.includes(languageCode)) {
+      return languageCode;
+  }
+
+  console.log(`Language '${languageCode}' is not supported by LORIS.`);
+  return 'en';
+}
+
+function getLanguageCodeFromBrowser() {
+  const languageCode = localStorage.getItem('lang');
+  if (languageCode !== null) {
+      return languageCode;
+  }
+
+  return navigator.language.split('-')[0];
+}
+
+interface Language {
+  code: string;
+  name: string;
+}
+
+interface LanguageProps {
+  language: Language;
+  appLanguageCode: string;
+  setAppLanguageCode: (languageCode: string) => void
+}
+
+function Language(props: LanguageProps) {
+  return (
+    <button
+        className={props.appLanguageCode === props.language.code ? 'selected' : ''}
+        onClick={() => props.setAppLanguageCode(props.language.code)}
+    >
+      {props.language.name}
+    </button>
+  );
+}
+
+const languages = [
+  {code: 'en', name: 'English'},
+  {code: 'fr', name: 'Français'},
+];
+
+function LanguageMenu() {
+  const {i18n} = useTranslation();
+  const [languageCode, setLanguageCode] = useState<string>('en');
+
+  const setAppLanguageCode = (languageCode: string) => {
+    languageCode = filterSupportedLanguageCode(languageCode);
+    localStorage.setItem('lang', languageCode);
+    i18n.changeLanguage(languageCode);
+    setLanguageCode(languageCode);
+  };
+
+  useEffect(() => {
+      setAppLanguageCode(getLanguageCodeFromBrowser());
```

Comment:
Not a big deal but this could be cleaner if this effect (and all the related state etc) were refactored into a `useLanguage` hook.

A slightly bigger deal: this should prioritize the user preference for the current user from LORIS.

Reply:
I agree with the `useLanguage` hook, I'll apply it.

Regarding the LORIS user prefered language, should it be in this PR ? I kept this one minimal for now (and will have other priorities in the coming days).

Reply:
If you handle it the same way as the getAvailableLanguages change I don't think it should be a large change? It's just a check before the browser check in the useLanguages hook.

Reply:
Hhhhm, I guess just reading the user language should indeed not be too big of a task. I was thinking of also updating the user preferred language when they change the setting in the dropdown but I guess that can wait for later since the setting can already be changed in the user config.